### PR TITLE
Support parallel union on universal tables [#127031149]

### PR DESCRIPTION
**Pull request by hsyuan**
For queries like this:
SELECT 1 UNION ALL SELECT * FROM tbl;
INSERT INTO tbl VALUES (11),(12),(13);

Now Orca support parallel union for these queries.

### CODE REVIEW

**Comment by hsyuan** [comment]
@vraghavan78 @d @cramja please take a look.

File: libgpopt/src/base/CDistributionSpecUniversal.cpp [hunk 1]
```diff
@@ -32,6 +33,14 @@ namespace gpopt
 			return false;
 		}
 
+		// universal distribution does not satisfy duplicate-sensitive
+		// strict hash distributions, which is only used in parallel union
+		if (CDistributionSpec::EdtStrictHashed == pds->Edt() &&
+			(CDistributionSpecStrictHashed::PdsConvert(pds))->FDuplicateSensitive())
```
Comment (d) [inline libgpopt/src/base/CDistributionSpecUniversal.cpp]: Either this `FDuplicateSensitive` is unnecessary, or we are missing a test. After removing the check, none of your tests broke.

File: data/dxl/minidump/CPhysicalParallelUnionAllTest/ParallelUnion-ConstTable.mdp [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+-->
+<dxl:DXLMessage xmlns:dxl="http://greenplum.com/dxl/2010/12/">
+  <dxl:Thread Id="0">
+    <dxl:OptimizerConfig>
+      <dxl:EnumeratorConfig Id="0" PlanSamples="0" CostThreshold="0"/>
+      <dxl:StatisticsConfig DampingFactorFilter="0.750000" DampingFactorJoin="0.010000" DampingFactorGroupBy="0.750000"/>
+      <dxl:CTEConfig CTEInliningCutoff="0"/>
+      <dxl:CostModelConfig CostModelType="1" SegmentsForCosting="3"/>
+      <dxl:Hint MinNumOfPartsToRequireSortOnInsert="2147483647" JoinArityForAssociativityCommutativity="2147483647" ArrayExpansionThreshold="25"/>
+      <dxl:TraceFlags Value="101001,102120,103001,103014,103015,103022,103025,104004,104005,105000"/>
```
Comment (d) [inline data/dxl/minidump/CPhysicalParallelUnionAllTest/ParallelUnion-ConstTable.mdp]: please don't include the flags `101001` here, it makes the test very noisy

Comment (hsyuan) [inline data/dxl/minidump/CPhysicalParallelUnionAllTest/ParallelUnion-ConstTable.mdp]: make sense

File: libgpopt/include/gpopt/operators/CPhysicalParallelUnionAll.h [hunk 3]
```diff
@@ -34,6 +34,15 @@ namespace gpopt
 			virtual CEnfdProp::EPropEnforcingType
 			EpetDistribution(CExpressionHandle &exprhdl, const CEnfdDistribution *ped) const;
 
+			virtual
+			CEnfdDistribution::EDistributionMatching Edm
```
Comment (d) [inline libgpopt/include/gpopt/operators/CPhysicalParallelUnionAll.h]: The function declaration in header can have the parameter names. The compiler only yells at unused parameter names in the definition (.cpp)

**Comment by d** [comment]
Does it make sense to clean up the inlined definitions in `CPhysical.h` while we are there?

**Pull request closed without merging**